test/integration/TokenAirdropIntegrationTest.js

```diff
@@ -0,0 +1,538 @@
+                    env.operatorId,
+                    -INITIAL_SUPPLY,
+                )
+                .addTokenTransfer(
+                    tokenWithFeeId,
+                    senderAccountId,
+                    INITIAL_SUPPLY,
+                )
+                .addNftTransfer(
+                    nftTokenId,
+                    serials[0],
+                    env.operatorId,
+                    senderAccountId,
+                )
+                .execute(env.client)
+        ).getReceipt(env.client);
+
+        await (
+            await new TransferTransaction()
+                .addTokenTransfer(feeTokenId, env.operatorId, -INITIAL_SUPPLY)
+                .addTokenTransfer(feeTokenId, senderAccountId, INITIAL_SUPPLY)
+                .execute(env.client)
+        ).getReceipt(env.client);
+
+        await (
+            await (
+                await new TokenAirdropTransaction()
+                    .addTokenTransfer(
+                        tokenWithFeeId,
+                        receiverId,
+                        INITIAL_SUPPLY,
+                    )
+                    .addTokenTransfer(
+                        tokenWithFeeId,
+                        senderAccountId,
+                        -INITIAL_SUPPLY,
+                    )
+                    .addNftTransfer(
+                        nftTokenId,
+                        serials[0],
+                        senderAccountId,
+                        receiverId,
+                    )
+                    .freezeWith(env.client)
+                    .sign(senderPrivateKey)
+            ).execute(env.client)
+        ).getReceipt(env.client);
+
+        const operatorBalance = await new AccountBalanceQuery()
+            .setAccountId(env.operatorId)
+            .execute(env.client);
+
+        // check if fees are collected
+        const DISTINCT_TRANSACTIONS = 2;
+        expect(operatorBalance.tokens.get(tokenWithFeeId).toInt()).to.be.eq(0);
+        expect(operatorBalance.tokens.get(feeTokenId).toInt()).to.be.eq(
+            DISTINCT_TRANSACTIONS,
+        );
+
+        const receiverBalance = await new AccountBalanceQuery()
+            .setAccountId(receiverId)
+            .execute(env.client);
+        expect(receiverBalance.tokens.get(tokenWithFeeId).toInt()).to.be.eq(
+            INITIAL_SUPPLY,
+        );
+
+        const senderBalance = await new AccountBalanceQuery()
+            .setAccountId(senderAccountId)
+            .execute(env.client);
+        expect(senderBalance.tokens.get(tokenWithFeeId).toInt()).to.be.eq(0);
+        expect(senderBalance.tokens.get(feeTokenId).toInt()).to.be.eq(
+            INITIAL_SUPPLY - DISTINCT_TRANSACTIONS * FEE_AMOUNT,
+        );
+        expect(senderBalance.tokens.get(nftTokenId).toInt()).to.be.eq(0);
+        expect(receiverBalance.tokens.get(nftTokenId).toInt()).to.be.eq(1);
+    });
+
+    it("should not airdrop with receiver sig set to true", async function () {
```

Comment:
This scenario was changed in the last couple of commits in services, so now it's possible to airdrop with receiver sig = true